network-providers/transaction_watcher.md

```diff
@@ -0,0 +1,31 @@
+## TransactionWatcher
```

Comment:
I'd vote to rename it into `TransactionAwaiter`. It's methods are `await_*`. Additionally `watching` makes us think of `continuously watching` - however, in practice, we don't want to watch a tx forever.

Reply:
I've now seen the other comment. All right, let's keep it as it is.

Reply:
I think you are right. Renamed to `TransactionAwaiter`.